recipe/meta.yaml

```diff
@@ -20,15 +20,15 @@ requirements:
     - nlopt
     - boost     1.63.*
     - numpy     x.x
-    - dill
+    - cloudpickle
```

Comment:
We had some issues related to the serialization of optimisation problems, see https://github.com/esa/pagmo2/issues/106

We tried to switch to cloudpickle and it ran our test suite successfully while also running the use cases mentioned in the report without further configuration, so we decided for the switch.

We might consider in the future to make the serialization backend a configurable option. For our uses cases, the interfaces exposed by dill, cloudpickle and pickle are all very compatible and it's easy to switch from one library to the other.